cmd/cluster/pullsecret.go

```diff
@@ -0,0 +1,90 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func newCmdPullSecret(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	return &cobra.Command{
+		Use:               "pull-secret",
+		Short:             "Checks if the pull secret email matches the owner email",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ValidatePullSecret(args[0], kubeCli, flags))
+		},
+	}
+}
+
+func ValidatePullSecret(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
+	ocm := utils.CreateConnection()
+	defer func() {
+		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
+			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
+		}
+	}()
+
+	fmt.Println("Checking if pull secret email matches user email")
+
+	flags.Impersonate = pointer.StringPtr("backplane-cluster-admin")
+	secret := &corev1.Secret{}
+	err := kubeCli.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-config", Name: "pull-secret"}, secret)
+	if err != nil {
+		return err
+	}
+
+	type auth struct {
+		Email string `json:"email"`
+	}
+
+	type auths struct {
+		Cloudopenshiftcom auth `json:"cloud.openshift.com"`
+	}
+
+	type response struct {
+		Auths auths `json:"auths"`
+	}
+
+	dockerConfigJson := response{}
+	err = json.Unmarshal(secret.Data[".dockerconfigjson"], &dockerConfigJson)
+	if err != nil {
+		return err
+	}
+
+	clusterPullSecretEmail := dockerConfigJson.Auths.Cloudopenshiftcom.Email
+	if clusterPullSecretEmail == "" {
+		fmt.Printf("%v\n%v\n%v\n",
+			"Couldn't extract email address from pull secret for cloud.openshift.com",
+			"This can mean the pull secret is misconfigured. Please verify the pull secret manually:",
+			"  oc get secret -n openshift-config pull-secret -o json | jq -r '.data.\".dockerconfigjson\"' | base64 -d")
```

Comment:
This command doesn't work for me with `jq-1.6` on Fedora.

If the following also works on Mac, can we update this suggested command to:
```
oc get secret -n openshift-config pull-secret -o json | jq -r '.data[".dockerconfigjson"]' | base64 -d
```

Reply:
For what it's worth, these outputs were copied [directly from the old check_banned_user script](https://github.com/openshift/ops-sop/blob/master/v4/utils/check_banned_user.sh#L70).

The command you shared also works on my mac, so I'll update to use that instead. Thanks for checking this!